evaluators/ragas_evaluator.py

```
@@ -159,54 +161,115 @@ def _evaluate_ragas_sync(self, question: str, answer: str, ground_truth: str, co
+            eval_contexts = contexts if contexts else [['相关上下文'] for _ in range(len(questions))]
+            
+            # 创建数据集
+            from datasets import Dataset
+            dataset = Dataset.from_dict({
+                'question': questions,
+                'answer': answers,
+                'ground_truth': ground_truths,
+                'contexts': eval_contexts
+            })
+            
+            # 使用Ragas批量评估（同步函数）
```

Ruff (0.12.2)

Comment contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF003)